crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -850,6 +862,25 @@ impl SpecializedMeshPipeline for MeshPipeline {
             TextureFormat::bevy_default()
         };
 
+        let mut targets = vec![Some(ColorTargetState {
+            format,
+            blend,
+            write_mask: ColorWrites::ALL,
+        })];
+
+        if key.contains(MeshPipelineKey::GPU_PICKING) {
+            shader_defs.push("GPU_PICKING".into());
+        }
+
+        if key.contains(MeshPipelineKey::MESH_ID_TEXTURE_TARGET) {
+            // we need to add the target even if the mesh isn't pickable
```

Comment:
I don't understand what this means. Doesn't `MESH_ID_TEXTURE_TARGET` means it _is_ pickable? (ie: has the `GpuPickingMesh` component)

If so, the comment contradicts the code.

Reply:
The issue is that if gpu picking is enabled, the mesh_id texture target needs to be added as an attachment to the entire render pass not just for the meshes that are pickable. So even if there's only a single mesh that is pickable the entire pipeline needs to be aware of the attachment.